src/compat/compat.h

```diff
@@ -808,4 +808,42 @@ RelationGetSmgr(Relation rel)
 #endif
 #endif
 
+/*
+ * PG16 replaces most aclcheck functions with a common object_aclcheck() function
```

Comment:
Will it be better to do the opposite and adopt the new interface for older PG versions, rather than sticking with the old going forward? If we adopt the new interface, it will allow us to more easily drop the backwards-compatibility stuff once we only support PG16 and greater.

In other words, can we provide `object_aclcheck()` for PG15, 14, etc. like this:

```
AclResult
object_aclcheck(Oid classid, Oid objectid, AclMode mode)
{
      switch (classid)
      {
           case NamespaceRelationId:
                 return pg_namespace_aclcheck(objectid, mode);
           case TableSpaceRelationId:
                 ...
     }
}
```

Reply:
Done. I have updated the PR based your comment.